test/framework/deployment_helpers.go

```diff
@@ -121,6 +132,22 @@ func WatchDeploymentLogs(ctx context.Context, input WatchDeploymentLogsInput) {
 		for _, container := range deployment.Spec.Template.Spec.Containers {
 			log.Logf("Creating log watcher for controller %s/%s, pod %s, container %s", input.Deployment.Namespace, input.Deployment.Name, pod.Name, container.Name)
 
+			logMetadataFile := filepath.Clean(path.Join(input.LogPath, input.Deployment.Name, pod.Name, container.Name+".json"))
```

Comment:
The idea here is to store the metadata for the logs already here when we have all the data, so we don't have to calculate it later when syncing the logs to loki

(the metadata is ~ equivalent to the metadata promtail is producing)

Reply:
I'm fine but wondering if we can make it more clear what this file contains by adding -metadata suffix

Reply:
Sounds good to me. We just have to get consensus on the file name. The only requirement I have is that I have to be able to calculate the metadata file name based on the log file name.

Reply:
Chose one